src/NumericalAlgorithms/Convergence/HasConverged.cpp

```diff
@@ -67,6 +71,9 @@ std::ostream& operator<<(std::ostream& os,
                          const HasConverged& has_converged) noexcept {
   if (has_converged) {
     switch (has_converged.reason()) {
+      case Reason::NumIterations:
+        return os << "Reached the target number of iterations ("
+                  << has_converged.criteria_.max_iterations << ").";
```

Comment:
this prints `max_iterations` which seems incorrect. However, I'm not terribly clear on the distinction between target number of iterations and maximum number of iterations

Reply:
added some more docs in a fixup